third_party/move/move-compiler/src/parser/ast.rs

```diff
@@ -518,6 +518,12 @@ pub type Bind = Spanned<Bind_>;
 // b1, ..., bn
 pub type BindList = Spanned<Vec<Bind>>;
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct TypedBind_(pub Bind, pub Option<Type>);
```

Comment:
We should have a test case where some parameters of a lambda are explicitly typed, but some are not (e.g., `|x: u64, y| { x + y }`). Do we want this to be legal?

Reply:
Added a test case at the end of `third_party/move/move-compiler-v2/tests/checking/typing/lambda_typed.move` and new test `third_party/move/move-compiler-v2/transactional-tests/tests/inlining/eval_ignored_param_some_typed.move`.